test/core/TestingFunctionMacro_UnitTests.cmake

```diff
@@ -978,6 +995,72 @@ FUNCTION(UNITTEST_TRIBITS_ADD_TEST_BASIC)
   UNITTEST_NOT_HAS_SUBSTR_CONST(TRIBITS_SET_TEST_PROPERTIES_INPUT
     "DISABLED")
 
+  MESSAGE("<fullTestName>_SET_RUN_SERIAL=ON")
+  SET(TRIBITS_SET_TEST_PROPERTIES_CAPTURE_INPUT ON)
+  SET(${PACKEXEN}_SET_RUN_SERIAL ON)
+  TRIBITS_ADD_TEST( ${EXEN}
+    ADDED_TESTS_NAMES_OUT ${EXEN}_TEST_NAMES )
+  UNITTEST_COMPARE_CONST( ${EXEN}_TEST_NAMES
+    "PackageA_SomeExec" )
+  UNITTEST_COMPARE_CONST(
+    TRIBITS_ADD_TEST_ADD_TEST_INPUT
+    "${PACKEXEN};${PACKEXEN_PATH}"
+    )
+  UNITTEST_COMPARE_CONST( MESSAGE_WRAPPER_INPUT
+    "-- PackageA_SomeExec: Added test (BASIC, PROCESSORS=1, RUN_SERIAL)!" )
+  UNITTEST_HAS_SUBSTR_CONST(TRIBITS_SET_TEST_PROPERTIES_INPUT
+    "RUN_SERIAL")
+
+  MESSAGE("<fullTestName>_SET_RUN_SERIAL=ON with NAME_POSTFIX and POSTFIX_AND_ARGS_<IDX>")
+  SET(TRIBITS_SET_TEST_PROPERTIES_CAPTURE_INPUT ON)
+  SET(THIS_TEST_NAME ${PACKEXEN}_mypostfix_argpostfix0)
+  SET(${THIS_TEST_NAME}_SET_DISABLED_AND_MSG "Disabled because of H and I")
+  TRIBITS_ADD_TEST( ${EXEN} NAME_POSTFIX mypostfix POSTFIX_AND_ARGS_0 argpostfix0 arg0
+    ADDED_TESTS_NAMES_OUT ${EXEN}_TEST_NAMES )
+  UNITTEST_COMPARE_CONST( ${EXEN}_TEST_NAMES
+    "${THIS_TEST_NAME}" )
+  UNITTEST_COMPARE_CONST(
+    TRIBITS_ADD_TEST_ADD_TEST_INPUT
+    "${THIS_TEST_NAME};${PACKEXEN_PATH};arg0"
+    )
+  UNITTEST_COMPARE_CONST( MESSAGE_WRAPPER_INPUT
+    "-- ${THIS_TEST_NAME}: Added test (BASIC, PROCESSORS=1, DISABLED)!;--  => Reason DISABLED: Disabled because of H and I" )
+  UNITTEST_HAS_SUBSTR_CONST(TRIBITS_SET_TEST_PROPERTIES_INPUT
+    "DISABLED;ON")
+
+  MESSAGE("Set RUN_SERIAL input argument")
+  SET(TRIBITS_SET_TEST_PROPERTIES_CAPTURE_INPUT ON)
+  SET(${PACKEXEN}_SET_RUN_SERIAL "")
+  TRIBITS_ADD_TEST( ${EXEN} RUN_SERIAL
+    ADDED_TESTS_NAMES_OUT ${EXEN}_TEST_NAMES )
+  UNITTEST_COMPARE_CONST( ${EXEN}_TEST_NAMES
+    "PackageA_SomeExec" )
+  UNITTEST_COMPARE_CONST(
+    TRIBITS_ADD_TEST_ADD_TEST_INPUT
+    "${PACKEXEN};${PACKEXEN_PATH}"
+    )
+  UNITTEST_COMPARE_CONST( MESSAGE_WRAPPER_INPUT
+    "-- PackageA_SomeExec: Added test (BASIC, PROCESSORS=1, RUN_SERIAL)!" )
+  UNITTEST_HAS_SUBSTR_CONST(TRIBITS_SET_TEST_PROPERTIES_INPUT
+    "RUN_SERIAL")
+
+
+  MESSAGE("Set RUN_SERIAL input argument but set <fullTestName>_SET_RUN_SERIAL=OFF")
+  SET(TRIBITS_SET_TEST_PROPERTIES_CAPTURE_INPUT ON)
+  SET(${PACKEXEN}_SET_RUN_SERIAL OFF)
+  TRIBITS_ADD_TEST( ${EXEN} RUN_SERIAL
+    ADDED_TESTS_NAMES_OUT ${EXEN}_TEST_NAMES )
+  UNITTEST_COMPARE_CONST( ${EXEN}_TEST_NAMES
+    "PackageA_SomeExec" )
+  UNITTEST_COMPARE_CONST(
+    TRIBITS_ADD_TEST_ADD_TEST_INPUT
+    "${PACKEXEN};${PACKEXEN_PATH}"
+    )
+  UNITTEST_COMPARE_CONST( MESSAGE_WRAPPER_INPUT
+    "-- PackageA_SomeExec: Added test (BASIC, PROCESSORS=1)!" )
+  UNITTEST_NOT_HAS_SUBSTR_CONST(TRIBITS_SET_TEST_PROPERTIES_INPUT
+    "RUN_SERIAL")
```

Comment:
`RUN_SERIAL;OFF`?

Reply:
If the `RUN_SERIAL` property is not set, it defaults to OFF.  IMHO, it is better to not set it as `RUN_SERIAL OFF` in this case.